priv/perf/apps/load_test/lib/application.ex

```diff
@@ -15,10 +19,38 @@ defmodule LoadTest.Application do
         max_connections: max_connections
       )
 
+    :ok = start_services()
+
     {:ok, self()}
   end
 
   def stop(_app) do
     :hackney_pool.stop_pool(:perf_pool)
   end
+
+  defp start_services() do
+    {:ok, _} =
+      Supervisor.start_link([{LoadTest.Service.NonceTracker, []}], name: LoadTest.Supervisor, strategy: :one_for_one)
+
+    faucet_config = fetch_faucet_config()
+    # not started under supervisor as it creates and funds a new Ethereum account on each start
+    {:ok, _} = LoadTest.Service.Faucet.start_link(faucet_config)
+    :ok
+  end
+
+  defp fetch_faucet_config() do
+    faucet_opt =
+      case Application.fetch_env(:load_test, :faucet_account) do
+        {:ok, %{priv: priv}} ->
+          {:ok, faucet_account} = priv |> Encoding.to_binary() |> Account.new()
+          [faucet: faucet_account]
+
+        :error ->
```

Comment:
I think this case is irrelevant, you should be more aggressive.
```
{:ok, %{priv: priv}} = Application.fetch_env(:load_test, :faucet_account)
```

Reply:
It's releveant. We don't need faucet account to be set for running smoke test or simply running with dev geth.

Reply:
ah, so this isn't error handling (as it looks) but different way of setting a faucet? if that's the case, I think this needs better handling so that it's evident what the code implies.

Reply:
I clarified that a bit in the code, I hope it helps. If you think of handling it in a different way, let me know.

Reply:
t seems liek you don't need an account address at all.
```
faucet_private_key: "0x70ad9d48f90430607a340ea2e00495e1f84d2c50a4a3df0917e6161045b32378"
  
```
should be enough, 
maybe you can pull this out with
```
case Application.get_env(:load_test, :faucet_private_key, []) do
[] -> []
private_key = [faucet_private_key: faucet_private_key]
```
and yeah, `faucet ` should be renamed to `faucet_private_key`, so that we know what it holds without inspecting.